test/e2e/gardener/shoot/internal/node/node.go

```diff
@@ -35,79 +34,81 @@ const waitForCSINodeAnnotation = v1beta1constants.AnnotationPrefixWaitForCSINode
-	patchCSINodeObjectWithRequiredDriver(ctx, shootClient)
-
-	By("Verify node-critical components not ready taint is removed")
-	EventuallyWithOffset(1, func(g Gomega) []corev1.Taint {
-		g.Expect(shootClient.Get(ctx, client.ObjectKeyFromObject(node), node)).To(Succeed())
-		return node.Spec.Taints
-	}).WithContext(ctx).WithTimeout(10 * time.Minute).ShouldNot(ContainElement(corev1.Taint{
-		Key:    v1beta1constants.TaintNodeCriticalComponentsNotReady,
-		Effect: corev1.TaintEffectNoSchedule,
-	}))
-
-	cleanupNodeCriticalManagedResource(ctx, seedClient, technicalID, nodeCriticalDaemonSetName)
-	cleanupNodeCriticalManagedResource(ctx, seedClient, technicalID, csiNodeDaemonSetName)
+	It("Wait for new Node to be created", func(ctx SpecContext) {
+		Eventually(func(g Gomega) {
+			nodeList := &corev1.NodeList{}
+			g.Expect(s.ShootClient.List(ctx, nodeList)).To(Succeed())
+			g.Expect(nodeList.Items).NotTo(BeEmpty(), "new Node should be created")
+			node = &nodeList.Items[0]
+		}).Should(Succeed())
+	}, SpecTimeout(10*time.Minute))
+
+	It("Verify node-critical components not ready taint is present", func(ctx SpecContext) {
+		Consistently(func(g Gomega) []corev1.Taint {
+			g.Expect(s.ShootClient.Get(ctx, client.ObjectKeyFromObject(node), node)).To(Succeed())
+			return node.Spec.Taints
+		}).Should(ContainElement(corev1.Taint{
+			Key:    v1beta1constants.TaintNodeCriticalComponentsNotReady,
+			Effect: corev1.TaintEffectNoSchedule,
+		}))
+	}, SpecTimeout(10*time.Minute))
+
+	It("Update ManagedResources for shoot with working node-critical component", func(ctx SpecContext) {
+		// Use a container image that is already cached
+		image, err := imagevector.Containers().FindImage(imagevector.ContainerImageNamePauseContainer)
+		Expect(err).To(Succeed())
+
+		nodeCriticalDaemonSet := createOrUpdateNodeCriticalManagedResource(ctx, s.SeedClient, s.ShootClient, technicalID, nodeCriticalDaemonSetName, image.String(), false)
+		csiNodeDaemonSet := createOrUpdateNodeCriticalManagedResource(ctx, s.SeedClient, s.ShootClient, technicalID, csiNodeDaemonSetName, image.String(), true)
+
+		waitForDaemonSetToBecomeHealthy(ctx, s.ShootClient, nodeCriticalDaemonSet)
+		waitForDaemonSetToBecomeHealthy(ctx, s.ShootClient, csiNodeDaemonSet)
+	}, SpecTimeout(5*time.Minute))
+
+	It("Patch CSINode object to contain required driver", func(ctx SpecContext) {
+		patchCSINodeObjectWithRequiredDriver(ctx, s.ShootClient)
+	}, SpecTimeout(time.Minute))
+
+	It("Verify node-critical components not ready taint is removed", func(ctx SpecContext) {
+		Eventually(func(g Gomega) []corev1.Taint {
+			g.Expect(s.ShootClient.Get(ctx, client.ObjectKeyFromObject(node), node)).To(Succeed())
+			return node.Spec.Taints
+		}).ShouldNot(ContainElement(corev1.Taint{
+			Key:    v1beta1constants.TaintNodeCriticalComponentsNotReady,
+			Effect: corev1.TaintEffectNoSchedule,
+		}))
+	}, SpecTimeout(10*time.Minute))
+
+	It("Cleanup node-critical resources", func(ctx SpecContext) {
```

Comment:
I think, this `It` is no longer needed, now that we have a proper cleanup configured in `BeforeAll`.